src/rdb.c

```diff
@@ -1445,6 +1445,7 @@ int rdbSave(int req, char *filename, rdbSaveInfo *rsi) {
         stopSaving(0);
         return C_ERR;
     }
+    if (fsyncFileDir(filename) == -1) goto werr;
```

Comment:
same here `goto werr` is not valid at this point. the rename already succeeded (e.g. there's no point in `unlink(tmpfile)`.

Reply:
If we regard unlink(tmpfile) as a best effort attempt, after rename succeeded we could also try to `unlink(tmpfile)`(maybe meaningless) and ignore the possible errors. I commit the same error handling in config.c just now.